edb/edgeql/parser/parser.py

```diff
@@ -116,8 +117,14 @@ def get_exception(self, native_err, context, token=None):
                     else:
                         msg = f'Unexpected {token.text()!r}'
                 elif rule == 'for iterator':
-                    msg = (f"missing '{{' before {token.text()!r} "
-                           f"in a FOR query")
+                    msg = ("Complex expressions in a FOR query must be "
```

Comment:
```suggestion
                    msg = ("Complex expressions in a FOR iterator clause must be "
```

Reply:
Or, maybe "missing parentheses around complex expression in a FOR iterator clause"

Reply:
If we do the latter suggestion, should we also change "Statement used as an expression must be enclosed in a set of parentheses" to "Missing parentheses around statement used as expression"?

Reply:
Sure

Reply:
Hm, I think I prefer the first approach, actually. Fine to stick with that?